src/main/scala/tv/teads/wiremock/extension/FreeMarkerRenderer.scala

```diff
@@ -79,4 +83,44 @@ class FreeMarkerRenderer extends ResponseDefinitionTransformer {
       hash
   }
 
+  class FindFirstStringInArray(requestBody: JsonNode) extends TemplateMethodModelEx {
+    import freemarker.template.TemplateModelException
+
+    @tailrec
+    private def findChildNode(parent: JsonNode, fullPath: String): JsonNode = {
+      val childNodes = fullPath.split('.')
+      if (childNodes.length == 1)
+        parent.findPath(fullPath)
+      else findChildNode(parent.findPath(childNodes.head), childNodes.tail.mkString("."))
+    }
+
+    private def extractPathAndValueFromCondition(filteredChildCondition: String): (String, String) = {
+      val splitFilteredChildCondition = filteredChildCondition.split("==").map(_.trim)
+      if (splitFilteredChildCondition.length != 2) throw new TemplateModelException("Filtered child condition should be like this : car.color == red")
+      (splitFilteredChildCondition(0), splitFilteredChildCondition(1))
+    }
+
+    override def exec(arguments: util.List[_]): SimpleScalar = {
+      if (arguments.size() != 3) {
+        throw new TemplateModelException("Wrong arguments : 3 expected : array node, filtered child condition, wanted node )")
+      }
+      arguments.asScala.toList match {
```

Comment:
`.map(_.getAsString)`

Reply:
As i can only apply getAsString on a `SimpleScalar`, it would be like this i think : `arguments.asScala.toList.collect { case s: SimpleScalar => s.getAsString } match {`